src/components/SelectedItemBadge/SelectedItemBadge.tsx

```diff
@@ -71,158 +75,123 @@ export interface Props<PriorityItemValue, ChildrenItemValue>
-        /**
-         * Updates the dropdown state only if 'isOpen' is explicitly defined,
-         *  unintended state changes happen by mouse events cause 'isOpen' to become undefined
-         * */
-        if (isOpen !== undefined) {
-            setDropdownStates((prev) => ({ ...prev, [type]: isOpen }));
-        }
-    };
 
-    return (
-        <div {...rest} {...block({ ...rest })} ref={badgeRef}>
-            {hasPriorityList && (
-                <Dropdown<PriorityItem<PriorityItemValue>>
-                    button={
-                        <button
-                            {...elem('priorityButton', {
-                                isSelected: dropdownStates.priority,
-                            })}
-                            aria-label={priority.buttonLabel}
-                            disabled={isDisabled}
-                            type="button"
-                        >
-                            {renderPriorityIcon(priority.selectedItem?.priority, isDisabled)}
-                        </button>
-                    }
-                    additionalSelectProps={{
-                        onStateChange: (state) => toggleDropdown('priority', state.isOpen),
-                    }}
-                    onChange={(newPriorityItem) => priority.onChange(newPriorityItem)}
-                    placement="bottom-start"
-                    listClassName={styles.badgeDropdownList}
-                    refElement={badgeRef}
-                >
-                    {priority.list.map((item) => (
-                        <ListItem className={styles.badgeListItem} key={item.priority} value={item}>
-                            {renderPriorityIcon(item.priority)}
-                            <Text inline size="small">
-                                {item.label}
-                            </Text>
-                        </ListItem>
-                    ))}
-                </Dropdown>
-            )}
-
-            {children ? (
-                <Dropdown<ChildrenItemValue>
-                    button={
-                        <button
-                            {...elem('optionButton', {
-                                isSelected: dropdownStates.main,
-                            })}
-                            aria-label={buttonLabel}
-                            disabled={isDisabled}
-                            type="button"
-                        >
-                            <Text inline size="small" {...elem('valueText')}>
-                                {label}
-                            </Text>
-                            {additionalLabel && (
-                                <Text {...elem('optionText')} inline size="small">
-                                    {additionalLabel}
+export const SelectedItemBadge = React.forwardRef<HTMLDivElement, Props<string>>(
+    (
+        {
+            children,
+            label,
+            additionalLabel,
+            // isMultiSelect = false,
```

Comment:
Could we please update the stories for multiselection of the badge to ensure it works correctly with the new dropdown? Now it says: `Note: group selection and Select all is not correctly implemented in this example. We only showcasing the layout here.`